Backend/Tumultu/Application/FileVariants/Commands/CreateFileVariantCommand.cs

```diff
@@ -1,24 +1,54 @@
 ﻿using MediatR;
 using Tumultu.Application.Common.Interfaces;
+using Tumultu.Application.Users.Queries;
 using Tumultu.Domain.Entities;
+using Tumultu.Domain.Events;
 
 namespace Tumultu.Application.FileVariants.Commands;
 
 public record CreateFileVariantCommand : IRequest<Guid>
 {
+    public FileEntity? File { get; init; }
+    public DateTimeOffset? CreatedDate { get; init; }
+    public DateTimeOffset? DateModified { get; init; }
+    public string? Name { get; init; }
+    public Guid? UploadedBy { get; init; }
+
 }
 
 public class CreateFileVariantCommandHandler : IRequestHandler<CreateFileVariantCommand, Guid>
 {
     private readonly IWriteRepository<FileVariant, Guid> _writeRepository;
-
-    public CreateFileVariantCommandHandler(IWriteRepository<FileVariant, Guid> writeRepository)
+    private readonly IReadRepository<FileVariant, Guid> _readRepository;
+    private readonly IMediator _mediator;
+    public CreateFileVariantCommandHandler(IWriteRepository<FileVariant, Guid> writeRepository,
+                                           IReadRepository<FileVariant, Guid> readRepository,
+                                           IMediator mediator)
     {
         _writeRepository = writeRepository;
+        _readRepository = readRepository;
+        _mediator = mediator;
     }
 
-    public Task<Guid> Handle(CreateFileVariantCommand request, CancellationToken cancellationToken)
+    public async Task<Guid> Handle(CreateFileVariantCommand request, CancellationToken cancellationToken)
     {
-        throw new NotImplementedException();
+        var userQuery = new GetUserByIdQuery { Id = request.UploadedBy ?? default };
```

Comment:
I think we should not use the CQRS Queries from Commands. By the CQRS pattern the write and read part should be separated subsystems and I think the Command itself should be able to issue queries against the database because it is a part of update logic. 

Furthermore it concerns me how it is structured now and we should discuss it. Right now we are using ReadRepository (Dapper) and WriteRepository (EF Core) inside the Command. I feel like these are separated models and the WriteRepository itself should be actually able to issue queries against the database. One of main purposes of why CQRS is even used is to separate the read and write logic so they are easily scalable independently of each other.  Imagine a situation where we have 2 database nodes, one is read only and second is writable. CQRS allows for such physical distinction to be easily handled in the code. Due to that it should not be linked together logically in any sense. The example would be that the ReadRepositoy connects to let's say "Database 1" and does only reading from here. The WriteRepository connects to "Database 2" instead and does all the write operations including any reading needed with it. Right now it would get messy because we first query one database and based on the result do writing to the other.

In summary, I think we should implement WriteRepositories in a way that allows them to also read from database. Then ReadRepository would be used only in queries, and WriteRepositories would be used only for Commands. Also transaction management would be easier in case of write operations since we would be using singular type of repository in Commands (WriteRepository). I am not saying we will have to create multiple nodes database cluster but we might.  This is a topic for further discussion IMO.